napari/layers/utils/_slice_input.py

```diff
@@ -79,14 +81,31 @@ def data_indices(
 
         slice_world_to_data = world_to_data.set_slice(self.not_displayed)
         world_pts = [self.point[ax] for ax in self.not_displayed]
+
+        world_margin_left = [
+            self.point[ax] - self.margin_left[ax] for ax in self.not_displayed
+        ]
+        world_margin_right = [
+            self.point[ax] + self.margin_right[ax] for ax in self.not_displayed
+        ]
+
         data_pts = slice_world_to_data(world_pts)
+        data_margin_left = slice_world_to_data(world_margin_left)
+        data_margin_right = slice_world_to_data(world_margin_right)
+
         if round_index:
             # A round is taken to convert these values to slicing integers
             data_pts = np.round(data_pts).astype(int)
+            data_margin_left = np.round(data_margin_left).astype(int)
+            data_margin_right = np.round(data_margin_right).astype(int)
 
-        indices = [slice(None)] * self.ndim
+        indices = [[np.nan, np.nan, np.nan] for _ in range(self.ndim)]
```

Comment:
Does this make the return type of this method `Tuple[Tuple[Union[float, int], Union[float, int], Union[float, int]]]` now?

Reply:
Yes, this needs to be changed, and it's in fact one of the things where we might need to change a bit more deeply the code. Thing is, the return of this method used to be basically plug and play for slicing; now, however, it's a completely different 3-tuple with point, margin left and margin right.

It might be worth renaming this method as well.

Maybe indeed having a dataclass for this little piece of information is a good idea...